### ISSUE TITLE
Bug: Memory issue?

### ISSUE THREAD

**Issue by shane-breeze** [alphatwirl/alphatwirl-NanoAOD#3]
### Branch
https://github.com/shane-breeze/alphatwirl-NanoAOD/tree/dev

### Setup
Checkout the above branch as you normally would, initialise and update the submodules, source the setup script

### How to reproduce
Run the following at Imperial (or download the MET Run2016B root files):
```
python bin/trees2dataframes --components MET /vols/cms/sdb15/NanoAOD/NanoCrabXmasRun2016-03Feb2017/ --ncores 8 --n-files -1 -N 1000000
```

### Error
I've attached files with the stdout and stderr. The error comes after running on a significant number of events (near the end of 8 jobs that have run over 1 million events)

[out.txt](https://github.com/alphatwirl/alphatwirl-NanoAOD/files/1699540/out.txt)
[err.txt](https://github.com/alphatwirl/alphatwirl-NanoAOD/files/1699541/err.txt)

**Comment by shane-breeze** [comment]
Works fine with `--n-files 1`

**Comment by benkrikler** [comment]
@TaiSakuma Have you seen anything like this before?

**Comment by TaiSakuma** [comment]
No. I can only write my guess. Does that happen when a new file is opened in TChain? NanoAOD doesn't have the exact same branch structures in different files at least for triggers as far as I know. This error could happen for example if a branch in the 1st file in the chain that alphatwirl is reading doesn't exist in the 2nd file.
